samples/algorithms/MajoranaQubits/src/GateSet.qs

```diff
@@ -0,0 +1,65 @@
+/// A set of gates built upon the custom measurements
+/// provided by the hardware provider.
+///
+/// Source:
+///  [1] Surface code compilation via edge-disjoint paths
+///      https://arxiv.org/pdf/2110.11493
+
+/// Apply a CNOT gate to the given qubits.
+/// Source: [1] Figure 3.
+operation CNOT(control : Qubit, target : Qubit) : Unit {
+    // Prepare an ancilla qubit in the |+> state.
+    use ancilla = Qubit();
+    H(ancilla);
+
+    let a = Mzz(control, ancilla);
+    let b = Mxx(ancilla, target);
+    let c = Mx(ancilla);
+    if b == One {
+        Z(control);
+    }
+    if [a, c] == [One, One] {
+        X(target);
+    }
```

Comment:
I think this needs adjusting... unless the resource you are using defines it differently, I've always seen it used this way:

```suggestion
    let c = Mz(ancilla);
    if b == One {
        Z(control);
    }
    if a != c {
        X(target);
    }
```

Reply:
with that change, you'll probably need to add a reset of the ancilla before it goes out of scope.